docs: update examples and tips on commit statistics

ognis1205 opened the pull request:
## Description

This PR updates the example templates and adds a new tips & tricks section showcasing how to use `statistics` in `git-cliff` templates. It demonstrates how to include commit counts, durations, issue references, and other metrics in the generated changelog. Closes #1160 

## Motivation and Context

The usage of the statistics object is currently under-documented, which may prevent users from utilizing it effectively in their templates. This PR addresses that gap by adding practical examples and tips to help users take advantage of the available data in a meaningful way.

## How Has This Been Tested?

N/A

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other <!--- (provide information) -->

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have formatted the code with [rustfmt](https://github.com/rust-lang/rustfmt).
- [x] I checked the lints with [clippy](https://github.com/rust-lang/rust-clippy).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

<!--- Thank you for contributing to git-cliff! ⛰️  -->

orhun commented on examples/statistics.toml:
We need to update <https://git-cliff.org/docs/templating/examples> as well

The repo that is being used there is <https://github.com/orhun/git-cliff-readme-example>

orhun replied:
No that's not what I meant - we should update that examples page (to add this new template) based on the Git history in the linked repo.

ognis1205 commented on website/docs/tips-and-tricks.md:
````diff
@@ -102,3 +102,22 @@ To support unicode characters, use `xelatex` as PDF engine and a font family whi
+{%- if statistics.days_passed_since_last_release is defined %}
+	* {{ statistics.days_passed_since_last_release }} days passed between releases.
+{%- endif %}
+```
````
Sounds good — I’ll update it accordingly and include the example output as a collapsible section after the template. Thanks for the suggestion!

ognis1205 commented:
Hey @orhun,

I’ve addressed all the points that were raised in the review.
Let me know if there’s anything else you’d like me to revise or clarify.
Also, please review [#1151](https://github.com/orhun/git-cliff/pull/1151) together with this one when you have a chance.

Thanks!

orhun commented on examples/statistics.toml:
```diff
@@ -0,0 +1,79 @@
+# Order of commits in each group/release within the changelog.
+# Allowed values: newest, oldest
+sort_commits = "oldest"
+# An array of link parsers for extracting external references, and turning them into URLs, using regex.
+link_parsers = [
+    { pattern = "#(\\d+)", href = "https://github.com/orhun/git-cliff-readme-example/issues/$1"},
+]
```
I think we can remove this. It is irrelevant for other projects.

ognis1205 commented:
@orhun

I’ve addressed the changes suggested by in the review.

orhun reviewed (approved):
Thanks!

Moved this to context docs since it felt more appropriate there
